pkg/codegen/nodejs/gen.go

```diff
@@ -691,15 +691,12 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) error {
 	}
 
 	if len(secretProps) > 0 {
-		fmt.Fprintf(w, "        const secretOpts = { additionalSecretOutputs: [")
-		for i, sp := range secretProps {
-			if i > 0 {
-				fmt.Fprintf(w, ", ")
-			}
-			fmt.Fprintf(w, "%q", sp)
-		}
-		fmt.Fprintf(w, "] };\n")
-		fmt.Fprintf(w, "        opts = pulumi.mergeOptions(opts, secretOpts);\n")
+		fmt.Fprintf(w, "        // Always mark these fields as secret to avoid leaking sensitive values into the state.\n")
+		fmt.Fprintf(w, `        for (const key of ["%s"]) {`, strings.Join(secretProps, `", "`))
```

Comment:
If `secretProps` has more than one value (e.g. `[]string{"first", "second"}`), isn't this going to emit the following incorrect code?

```typescript
 for (const key of ["first, second"]) {
```

When instead, we really want:

```typescript
 for (const key of ["first", "second"]) {
```

Reply:
No, it works properly in both cases due to the `strings.Join` separator.
https://play.golang.org/p/GfsAhm0fHYL